python/ray/_private/event/event_logger.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import pathlib
+import json
+import random
+import string
+import socket
+import os
+import threading
+
+from typing import Dict
+from datetime import datetime
+
+from google.protobuf.json_format import MessageToDict
+
+from ray.core.generated.event_pb2 import Event
+
+
+def get_event_id():
+    return "".join([random.choice(string.hexdigits) for _ in range(36)])
+
+
+class EventLoggerAdapter:
+    def __init__(self, source: Event.SourceType, logger: logging.Logger):
+        """Adapter for the Python logger that's used to emit events.
+
+        This class is threas-safe.
+        """
+        self.logger = logger
+        # Aligned with `event.proto`'s `message Event``
+        self.source = source
+        self.source_hostname = socket.gethostname()
+        self.source_pid = os.getpid()
+
+        # The below fields must be protected by this lock.
+        self.lock = threading.Lock()
+        # {str -> str} typed dict
+        self.global_context = {}
```

Comment:
You don't really need a lock to assign a reference in Python.

Reply:
Not really, references are atomic in Python. You can remove the lock and it will work fine.

Reply:
Actually, I realized we have this when we publish events;

```
            for k, v in self.global_context.items():
                if v is not None and k is not None:
                    custom_fields[k] = v
```

And IIUC .items() is not atomic, so it's safer to keep it here.